src/BaseIO.hpp

```diff
@@ -328,6 +348,13 @@ class BaseIO
    * @brief Whether the file is currently open.
    */
   bool opened;
+
+  /**
+   * @brief Whether the file is in a mode where objects can
+   * be added or deleted. Note, this does not apply to the modification
+   * of raw data on already existing objects.
+   */
+  bool modifyObjects = true;
```

Comment:
I don't think we need the `modifyObjects` as a variable here. Instead, I think we can just have `canModifyObjects` return `true` in `BaseIO` and then `HDF5IO` and other backends can just overwrite `canModifyObjects`. The main reason is, that this will allow backends to implement logic to check for `canModifyObjects` instead of forcing the backend to maintain the status of the boolean flag.